composer.json

```diff
@@ -52,12 +60,24 @@
   "autoload": {
     "psr-4": {
       "AmpProject\\AmpWP\\": "src/"
-    }
+    },
+    "classmap": [
+      "includes/"
```

Comment:
How does this work? Does Composer just automatically glob all files in this directory to find all classes?

Reply:
Yes, `"classmap"` autoloading does exactly that.

Reply:
Yes, the classmap generator does indeed global all the files in the folders you provide and parses all the PHP it can find in there. It then creates a mapping between `$className => $filepath`. So you can use the classmap approach if you don't have a structured mapping between FQCNs and filepaths.

Also, the classmap that was generated is also used when you use the `--optimize` flag to generate the autolaoders. In that case, even the PSR-4 classes are scanned and mapped into the classmap, as the classmap is the fastest way to autoload.

On a production server, the classmap does usually not even need to load and parse. It will be an array that came from the OPCache, and the autoloading is a hashed lookup into that rarray.